python_modules/dagster/dagster/_model/decorator.py

```diff
@@ -201,6 +213,12 @@ def _asdict(self):
         return as_dict(self)
 
 
+class IHaveNew:
```

Comment:
I think this is the cleanest option but i'm not sure the best name for this

Reply:
`IHaveCustomNew` probably more accurate?

Reply:
i would love to not repeat "new" , 

```
@dagster_model_with_new
class Foo(IHaveCustomNew):
```

```
@dagster_model_with
class Foo(CustomNew):
```

```
@dagster_model_override
class Foo(IHaveCustomNew):
```